### PR TITLE
Fix macOS unity base path so that unity existence check works as intended

### PR DESCRIPTION
**Pull request by VioletXF**
#### Changes

- Fix a bug that existsSync(this.unityHubExecPath) always returns false

#### Checklist

- [x] Read the contribution [guide](https://github.com/game-ci/unity-builder/blob/main/CONTRIBUTING.md) and accept the
      [code](https://github.com/game-ci/unity-builder/blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Docs (If new inputs or outputs have been added or changes to behavior that should be documented. Please make
            a PR in the [documentation repo](https://github.com/game-ci/documentation))
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Comment by VioletXF** [comment]
force push: revert disabling -nographics (I just disabled this option for personal use and pushed by mistake. Sorry!)

File: src/model/platform-setup/setup-mac.ts [hunk 1]
```diff
@@ -110,17 +110,23 @@ class SetupMac {
...
-                                          --version ${buildParameters.editorVersion} \
-                                          --changeset ${unityChangeset.changeset} \
-                                          ${moduleArgument} \
-                                          --childModules `;
+    const execArguments: string[] = [
+      '--',
+      '--headless',
+      'install',
+      ...['--version', buildParameters.editorVersion],
+      ...['--changeset', unityChangeset.changeset],
```
Comment (webbertakken) [inline src/model/platform-setup/setup-mac.ts:120]: prefer template string over spreading two strings.

File: src/model/platform-setup/setup-mac.ts [hunk 2]
```diff
@@ -110,17 +110,23 @@ class SetupMac {
...
+      ...moduleArguments,
+      '--childModules',
+    ];
+
+    const escapedExecPath = this.unityHubExecPath.replace(/ /g, '\\ ');
 
     // Ignoring return code because the log seems to overflow the internal buffer which triggers
     // a false error
-    const errorCode = await exec(command, undefined, { silent, ignoreReturnCode: true });
+    const errorCode = await exec(escapedExecPath, execArguments, { silent, ignoreReturnCode: true });
```
Comment (webbertakken) [inline src/model/platform-setup/setup-mac.ts:129]: A bit of bikeshedding, but perhaps we could keep the argument names as `exec(command, arguments, {...`

**Review by webbertakken — APPROVED** [review]
LGTM

File: src/model/platform-setup/setup-mac.ts (hunk 1 above)
Comment (VioletXF) [reply src/model/platform-setup/setup-mac.ts:120]: I wasn't sure if exec handles whitespaces of arguments correctly, but I found it's OK after investigating its source code. I'll revert changing `getModuleParametersForTargetPlatform` too.

File: src/model/platform-setup/setup-mac.ts (hunk 2 above)
Comment (VioletXF) [reply src/model/platform-setup/setup-mac.ts:129]: I first tried `arguments`, but current linter config complained me not to use that name.

File: src/model/platform-setup/setup-mac.ts (hunk 1 above)
Comment (webbertakken) [reply src/model/platform-setup/setup-mac.ts:120]: I see.

**Comment by webbertakken** [comment]
I merged this but the pipeline in `main` is failing now.

**Comment by VioletXF** [comment]
@webbertakken I pushed fixes on #552. Sorry for making failing work :sob:

**Pull request merged**